shims.h

```diff
@@ -1,314 +1,380 @@
-typedef void* GO_OSSL_PARAM_PTR;
-typedef void* GO_CRYPTO_THREADID_PTR;
-typedef void* GO_EVP_SIGNATURE_PTR;
-typedef void* GO_DSA_PTR;
-typedef void* GO_EVP_KDF_PTR;
-typedef void* GO_EVP_KDF_CTX_PTR;
-
-// #include <openssl/md5.h>
-typedef void* GO_MD5_CTX_PTR;
-
-// #include <openssl/sha.h>
-typedef void* GO_SHA_CTX_PTR;
-
-// FOR_ALL_OPENSSL_FUNCTIONS is the list of all functions from libcrypto that are used in this package.
-// Forgetting to add a function here results in build failure with message reporting the function
-// that needs to be added.
-//
-// The purpose of FOR_ALL_OPENSSL_FUNCTIONS is to define all libcrypto functions
-// without depending on the openssl headers so it is easier to use this package
-// with an openssl version different that the one used at build time.
-//
-// The following macros may not be defined at this point,
-// they are not resolved here but just accumulated in FOR_ALL_OPENSSL_FUNCTIONS.
-//
-// DEFINEFUNC defines and loads openssl functions that can be directly called from Go as their signatures match
-// the OpenSSL API and do not require special logic.
-// The process will be aborted if the function can't be loaded.
-//
-// DEFINEFUNC_LEGACY_1_1 acts like DEFINEFUNC but only aborts the process if the function can't be loaded
-// when using 1.1.x. This indicates the function is required when using 1.1.x, but is unused when using later versions.
-// It also might not exist in later versions.
-//
-// DEFINEFUNC_LEGACY_1 acts like DEFINEFUNC but only aborts the process if the function can't be loaded
-// when using 1.x. This indicates the function is required when using 1.x, but is unused when using later versions.
-// It also might not exist in later versions.
-//
-// DEFINEFUNC_1_1_1 acts like DEFINEFUNC but only aborts the process if function can't be loaded
-// when using 1.1.1 or higher.
-//
-// DEFINEFUNC_3_0 acts like DEFINEFUNC but only aborts the process if function can't be loaded
-// when using 3.0.0 or higher.
-//
-// DEFINEFUNC_RENAMED_3_0 acts like DEFINEFUNC but tries to load the function using the new name when using >= 3.x
-// and the old name when using 1.x. In both cases the function will have the new name.
-//
-// DEFINEFUNC_VARIADIC_3_0 acts like DEFINEFUNC but creates an alias with a more specific signature.
-// This is necessary to call variadic functions (functions that accept a variable number of arguments)
-// because variadic functions are not directly compatible with cgo. By defining a cgo-compatible alias
-// for each desired signature, the C compiler handles the variadic arguments rather than cgo.
-// Variadic functions are the only known incompatibility of this kind.
-// If you use this macro for a different reason, consider renaming it to something more general first.
-// See https://github.com/golang/go/issues/975.
-// The process is aborted if the function can't be loaded when using 3.0.0 or higher.
-//
+// The following includes are used be the headercheck tool.
```

Comment:
```suggestion
// The following includes are used by the checkheader tool.
```

(I agree this is enough doc for these includes.)